### PR TITLE
Add a custom `Error` type for confy.

### PR DESCRIPTION
**Pull request by yuvallanger**
I talk about it on #11. I don't know if it works. It compiles.

### CODE REVIEW

**Comment by yuvallanger** [comment]
irc.mozilla.org's #rust-beginners said I shouldn't be returning a `std::boxed::Box<std::error::Error>`, instead just return a `ConfyError`, either with or without the help of the `failure` crate.

File: Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ serde = "1.0"
 toml = "0.4"
 directories = "0.10.0"
 serde_derive = { version = "1.0", optional = true }
+failure = "*"
```
Comment (Dylan-DPC) [inline Cargo.toml]: use the version instead of *

Comment (yuvallanger) [reply Cargo.toml]: Should I use something `failure = "0.1.5"` or `failure = "^0.1"` to let any upstream fixes of possible bugs filter through?

Comment (Dylan-DPC) [reply Cargo.toml]: `failure = "0.1"` should suffice since it is implicit `^0.1`

**Comment by yuvallanger** [comment]
I think I will just replace all the rest of the `unwrap()`s with more `ConfyError` values.
Please make sure the errors reported with those enum values are actually what those errors are.

File: Cargo.toml (hunk 1 above)
Comment (yuvallanger) [reply Cargo.toml]: > `failure = "0.1"` should suffice since it is implicit `^0.1`

I didn't know that. Thank you!

**Comment by yuvallanger** [comment]
I am not sure if I am doing it right, so please be careful in your review.

**Comment by Dylan-DPC** [comment]
Merging this. Thanks

**Comment by cosmicchipsocket** [comment]
Could we see this fix in a new crates.io release, please? I just hit an issue where a supposedly non-fatal parsing of the config file is causing the application to panic.

**Pull request merged**